packages/block-editor/src/store/actions.js

```diff
@@ -1197,7 +1203,7 @@ export const removeBlocks =
 		}
 
 		if ( selectPrevious ) {
-			dispatch.selectPreviousBlock( clientIds[ 0 ] );
```

Comment:
I think before selecting the previous block, we should check if selection is actually on or inside the block?

Reply:
Fixed in 39135d96a46ce6a51b3f907565cae5bee7f83e0f, but would be good to auto detect if selection should be adjusted